### ISSUE TITLE
Different scores in lightgbm_main and lightgbm_gridsearch

### ISSUE THREAD

**Issue by johannadevos** [mlp2018/fraud_detection#30]
I'm confused about something. When I run `lightgbm_gridsearch`, I'm getting the message that the best combination of parameter values achieved a score of 0.949. Then, those best parameter values are written to `optimal_lgbm_param_values.txt`. 

However, when I then run `lightgbm_main` with those optimized values, I'm only getting a score during cross-validation of 0.857. I would expect to get approximately the same score, since we're using the same parameter values.

Is this the same for other people as well, and if yes, where is the bug?

**Comment by aranas** [comment]
I can reproduce these numbers. Might this be due to some kind of "overfitting" during parameter search?

**Comment by johannadevos** [comment]
Thanks for that. Hmm, I don't know. The strange thing is that the main script uses the same parameter values that yielded the 0.949 score in the grid search, so if there would be any overfitting, you'd expect that it'd be the same in both scripts.

**Comment by aranas** [comment]
Okay, so indeed the fit_params variable was not passed to the fit function. But also, there seemed to be something strange happening with the cross-validation (first fold always yielded 0.5 roc score. I switched the cross-validation to the same stratifiedkfold we use in the lightgbm_main script but added a random seed for reproducibility, the values are the same now and also within the folds the values make more sense.

**Comment by johannadevos** [comment]
Nice work Sophie @aranas!
